File: 프로그래머스/1/12926. 시저 암호/시저 암호.js
function solution(s, n) {
 let answer = "";
    
    for (let i = 0; i < s.length; i++) {
        let char = s[i];

        if (char >= 'A' && char <= 'Z') {

            let newChar = String.fromCharCode(((char.charCodeAt(0) - 65 + n) % 26) + 65);
            answer += newChar;
        } else if (char >= 'a' && char <= 'z') {
   
            let newChar = String.fromCharCode(((char.charCodeAt(0) - 97 + n) % 26) + 97);
            answer += newChar;
        } else {
            answer += char;
        }
    }

    return answer;
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.java

import java.util.*;
import java.io.*;

class Solution {
    public String solution(String s, int n) {
        String answer = "";
        ArrayList upperList = new ArrayList();
        ArrayList lowerList = new ArrayList();


//      String [] sr = s.split("");

        for(int i= 65; i<=90 ;i++) {
            upperList.add((char) i);
        }


        for(int i = 65; i<=90 ; i++) {
            upperList.add((char) i);
        }

        for(int i= 97; i<=122 ; i++) {
            lowerList.add((char) i);
        }
        for(int i= 97; i<=122 ; i++) {
            lowerList.add((char) i);
        }


        for(int i= 0; i< s.length();i++) {
            int index =0;
            if(Character.isUpperCase(s.charAt(i))) {
                index = upperList.indexOf(s.charAt(i));
                answer+=upperList.get(index+n);

            } else if(Character.isLowerCase(s.charAt(i))) {
                index = lowerList.indexOf(s.charAt(i));
                answer+=lowerList.get(index+n);
            } else {
                answer+=" ";
            }


        }



        return answer;
    }
}
